ONSdigital/dp-search-reindex-api
# Feature/add failure scenarios

# What has changed




New component tests have been added to test failure scenarios and increase code coverage. The following component tests have been added:

1. Scenario: Job does not exist in the Job Store and a get request returns StatusNotFound
2. Scenario: No Jobs exist in the Job Store and a get request returns an empty list
3. Scenario: The connection to mongo DB is lost and a post request returns an internal server error
4. Scenario: Job does not exist in the Job Store and a put request, to update its number of tasks, returns StatusNotFound
5. Scenario: A put request fails to update the number of tasks because it contains an invalid value of count
6. Scenario: A put request fails to update the number of tasks because it contains a negative value of count
7. Scenario: The connection to mongo DB is lost and a put request returns an internal server error

The overall code coverage has increased from 49.5% to 55.7%

# How to review

The component tests can be run from the command line using this command: go test -component

However, to see the coverage from within GoLand, you can do as follows:
1. Run --> Edit Configurations..

2. Create a confirguation with the following settings (leave other settings blank):
Name: <anything you like> e.g. component test dp-search-reindex-api
Run on: Local machine
Test framework: gotest
Test kind: File
Files: <path to main_test.go> e.g. ~/dp-search-reindex-api/main_test.go
Working directory: ~/dp-search-reindex-api
Program arguments: -component
Module: dp-search-reindex-api

3. In the 'Project' window, on the left of the GoLand UI, right-click on main_test.go

4. Select More Run/Debug --> Run <name of your configuration> with Coverage

The component tests should then run in the 'Cover' window at the bottom of the UI and the coverage should be shown at the end.

# Who can review

Anyone but me
